### DISCUSSION TITLE
Full screen and double animation - iOS

### DISCUSSION THREAD

**Discussion by rgbaman** [timbru31/cordova-plugin-lottie-splashscreen#483] [General]
Not sure if this is a bug or a misconfiguration. When using the plugin with Ionic 6/Capacitor v4, it plays the animation twice, once in the bottom right corner and then secondly in the correct position. It also doesn't extend full screen, however it accepts the background color in full screen when I have changed it to red to test. (Demo attached).

![ezgif-3-c31bda84b7](https://user-images.githubusercontent.com/85580199/198520722-52d1b509-73a1-4cd4-a4ba-c5f032260ab8.gif)

My `capacitor.config.ts` file is as follows:

```
...
cordova : {
    preferences: {
      LottieFullScreen: "true",
      LottieHideAfterAnimationEnd: "false",
      LottieAnimationLocation: "public/assets/splash-circle.json",
    }
  },
...
```
There were numerous suggestions online that hiding the splash screen and then showing it again in the `app.component.ts` file should fix it, but not such luck with these settings:

```
private async showSplash(){
    await this.platform.ready().then(async () => {
      const lottie = (window as any).lottie;
      if (this.platform.is('ios') && lottie) {
        await lottie.splashscreen.hide();
        await lottie.splashscreen.show('public/assets/splash-circle.json', false);
      }
    });
  }
```
Do you have any suggestions?

**Comment by simbrams** [comment]
I have the same issue, the first frames of the lottie animation are played in the bottom right corner and then it goes full screen.

**Comment by simbrams** [comment]
I have managed to fix this issue by doing the following:

in `node_modules/cordova-plugin-lottie-splashscreen/plugin.xml` I have set `onload` to `false`
```xml
<feature name="LottieSplashScreen">
    <param name="ios-package" value="LottieSplashScreen" />
    <param name="onload" value="false" />
</feature>
```

Then in my `app.component.ts` I have this in the function called by the constructor:
```ts
async showSplash() {
    const lottie = (window as any).lottie;

    await lottie.splashscreen.hide();
    await lottie.splashscreen.show();
    ...
}
```

And finally in my `capacitor.config.json` :

```json
"cordova": {
      "preferences": {		
	      "LottieRemoteEnabled": "false",
	      "LottieAnimationLocationDark": "public/assets/splash-dark.json",
	      "LottieAnimationLocationLight": "public/assets/splash-light.json",
	      "LottieFullScreen": "true",
	      "LottieHideAfterAnimationEnd": "true",
	      "LottieBackgroundColorLight": "#ffffff",
	      "LottieBackgroundColorDark": "#000000"
      }
}
```

**Reply by iOznny** [reply]
@simbrams Thanks for your help, I set the onload parameter to false and it worked for me.

**Comment by timbru31** [comment]
I can only strongly advise **not to set onload to false**.
This defeats the whole purpose of lottie: running before your JS bundle is loaded correctly.

In most cases you should not call `.show()` again and just identify what's the correct way to hide your animation (e.g., after some seconds or a crucial part of your app is loaded)

**Reply by simbrams** [reply]
Well, curious why it works better that way.. 

I have completed by adding before `.show() `

`await this.platform.ready();`

And it seems to work well

**Reply by AsimDawrat** [reply]
@timbru31 any solution you propose other than this one, please?
